.github/actions/download/action.yml

```
@@ -26,7 +26,7 @@ runs:
         TARGET: ${{ inputs.path }}
         ARCHIVE: /tmp/downloads/${{ inputs.name }}.tar.zst
         SKIP_IF_DOES_NOT_EXIST: ${{ inputs.skip-if-does-not-exist }}
-        PREFIX: artifacts/${{ inputs.prefix || format('{0}/{1}', github.run_id, github.run_attempt) }}
+        PREFIX: artifacts/${{ inputs.prefix || format('{0}/{1}/{2}', github.sha, github.run_id, github.run_attempt) }}
```

We might want to use `github.event.pull_request.head.sha || github.sha` to make commit sha a bit more predictable for PRs.

By default, `actions/checkout` checkouts a phantom merge commit (your PR into the main), and then it builds and runs tests on it. But it's quite hard to find this commit sha if you know only your commit sha.

For example, for your commit https://github.com/neondatabase/neon/commit/ab1ad9d79cfc211a18f1db0119dd1387b146a207 it runs tests on https://github.com/neondatabase/neon/commit/e16f7fcded4e1142aee4478add6b86901a797e94.

So we can use a real commit sha instead of a phantom one (which makes things a bit less true, but much more convinient)

```suggestion
        PREFIX: artifacts/${{ inputs.prefix || format('{0}/{1}/{2}', github.event.pull_request.head.sha || github.sha, github.run_id, github.run_attempt) }}
```

The "issues" affect only builds in PRs; on the branches, it works pretty straightforwardly.